gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-healthcheck/src/main/java/io/gravitee/gateway/services/healthcheck/rule/EndpointRuleHandler.java

```diff
@@ -442,6 +445,16 @@ public void setNode(Node node) {
         this.node = node;
     }
 
+    public long getDelayMillis() {
+        CronTrigger expression = new CronTrigger(rule.schedule());
+        Date nextExecutionDate = expression.nextExecutionTime(new SimpleTriggerContext());
+        if (nextExecutionDate == null) { // NOSONAR nextExecutionDate is null if the trigger won't fire anymore
+            return -1;
+        }
+
+        return nextExecutionDate.getTime() - new Date().getTime();
+    }
```

Comment:
Not sure about this, at first sight, I would have put it in the `EndpointRuleCronHandler` but we also need it in the `HttpEndpointRuleHandler`, that's why I put it in the super class